internal/mode/static/manager.go

```diff
@@ -133,7 +133,7 @@ func StartManager(cfg config.Config) error {
 	}
 
 	// Ensure NGINX is running before registering metrics & starting the manager.
-	if err := ngxruntime.EnsureNginxRunning(ctx); err != nil {
+	if err := ngxruntime.EnsureNginxRunning(ctx, &ngxruntime.ProcessHandlerImpl{}); err != nil {
```

Comment:
I think I'd rather initialize a single processHandler, make the `EnsureNginxRunning` function a method for that object, and then reuse the processHandler in the runtime manager initialization.

Reply:
will take a close look into this one

Reply:
@sjberman I have added an update for this one, and if thats okey for you I think whole pr is ready for another review, 
thanks thanks hehe !!